kernel/arch/dreamcast/hardware/maple/maple_driver.c

```diff
@@ -104,29 +121,33 @@ int maple_driver_attach(maple_frame_t *det) {
     return 0;
 }
 
-static void maple_detach(maple_device_t *dev) {
+/* Detach an attached maple device */
+int maple_driver_detach(int p, int u) {
+    maple_device_t *dev = maple_enum_dev(p, u);
+
+    if(!dev)
+        return -1;
+
     if(dev->drv && dev->drv->detach)
         dev->drv->detach(dev->drv, dev);
 
+    dev->valid = false;
```

Comment:
I was trying to preserve the previous methodology of setting it, but since we don't have any driver detach functions currently, it shouldn't make a difference (the only reason I'd care is if some drivers had detach that worked against `dev->valid`). That should help reduce the possibility for clashes.